### DISCUSSION TITLE
EdgeX Docker deployment keeps restarting

### DISCUSSION THREAD

**Discussion by sounak-1989** [edgexfoundry/community#17] [General]
Hi All,

I am running Ubuntu Server 22.04 LTS in a VirtualBox inside Windows 10. I am able to successfully run docker containers in it. But when I am trying to deploy EdgeX using the EdgeX-Compose repository with the no-secty command all the core services keep restarting. Only services running properly are consul and redis.
When I try to see the docker logs for core-data I get the following.

level=INFO ts=2022-10-11T22:02:04.350549681Z app=core-data source=config.go:392 msg="Loaded service configuration from /res/configuration.toml"
level=INFO ts=2022-10-11T22:02:04.351128515Z app=core-data source=variables.go:371 msg="Variables override of 'Databases.Primary.Host' by environment variable: DATABASES_PRIMARY_HOST=edgex-redis"
level=INFO ts=2022-10-11T22:02:04.351152214Z app=core-data source=variables.go:371 msg="Variables override of 'MessageQueue.Host' by environment variable: MESSAGEQUEUE_HOST=edgex-redis"
level=INFO ts=2022-10-11T22:02:04.351158284Z app=core-data source=variables.go:371 msg="Variables override of 'Registry.Host' by environment variable: REGISTRY_HOST=edgex-core-consul"
level=INFO ts=2022-10-11T22:02:04.351162553Z app=core-data source=variables.go:371 msg="Variables override of 'Service.Host' by environment variable: SERVICE_HOST=edgex-core-data"
level=INFO ts=2022-10-11T22:02:04.351262379Z app=core-data source=config.go:161 msg="Using Configuration Provider access token of length 0"
level=INFO ts=2022-10-11T22:02:04.351268409Z app=core-data source=config.go:367 msg="Using Configuration provider (consul) from: http://edgex-core-consul:8500 with base path of edgex/core/2.0/core-data"
level=ERROR ts=2022-10-11T22:02:14.357025181Z app=core-data source=config.go:180 msg="configuration provider is not available"
level=ERROR ts=2022-10-11T22:02:25.35845714Z app=core-data source=config.go:180 msg="configuration provider is not available"
level=ERROR ts=2022-10-11T22:02:36.363766648Z app=core-data source=config.go:180 msg="configuration provider is not available"
level=ERROR ts=2022-10-11T22:02:47.368369597Z app=core-data source=config.go:180 msg="configuration provider is not available"
level=ERROR ts=2022-10-11T22:02:58.373320549Z app=core-data source=config.go:180 msg="configuration provider is not available"
level=ERROR ts=2022-10-11T22:03:09.374630023Z app=core-data source=config.go:180 msg="configuration provider is not available"
level=INFO ts=2022-10-11T22:03:10.375093844Z app=core-data source=registry.go:61 msg="Using Registry access token of length 0"
level=INFO ts=2022-10-11T22:03:10.375133705Z app=core-data source=registry.go:85 msg="Using Registry (consul) from http://edgex-core-consul:8500"
level=ERROR ts=2022-10-11T22:03:10.375160111Z app=core-data source=bootstrap.go:46 msg="unable to register with Registry in allotted time"

Any help is appreciated. Thanks.

**Comment by lenny-goodell** [comment]
Please attach the compose file you are using.

Can you browse successfully to localhost:8500 and interact with the Consul UI?
There is an obvious networking issue between the services and Consul. The url `http://edgex-core-consul:8500/` is correct for within the docker network.

**Reply by sounak-1989** [reply]
I am not able to access consul using the name of the service. but if I use the ip or localhost then I am able to access it.

**Reply by lenny-goodell** [reply]
Correct, the name is the docker host name only accessible from the docker network created when you run docker compose.

**Reply by lenny-goodell** [reply]
Exactly how are you running the compose file?

**Reply by sounak-1989** [reply]
I am using the compose builder to generate a no-secty docker-compose and then run docker-compose up -d command

**Comment by sounak-1989** [comment]
Here's the compose file:

networks:
  edgex-network:
    driver: bridge
services:
  app-service-rules:
    container_name: edgex-app-rules-engine
    depends_on:
      consul:
        condition: service_started
      data:
        condition: service_started
    environment:
      CLIENTS_CORE_COMMAND_HOST: edgex-core-command
      CLIENTS_CORE_DATA_HOST: edgex-core-data
      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
      CLIENTS_SUPPORT_NOTIFICATIONS_HOST: edgex-support-notifications
      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
      DATABASES_PRIMARY_HOST: edgex-redis
      EDGEX_PROFILE: rules-engine
      EDGEX_SECURITY_SECRET_STORE: "false"
      MESSAGEQUEUE_HOST: edgex-redis
      REGISTRY_HOST: edgex-core-consul
      SERVICE_HOST: edgex-app-rules-engine
      TRIGGER_EDGEXMESSAGEBUS_PUBLISHHOST_HOST: edgex-redis
      TRIGGER_EDGEXMESSAGEBUS_SUBSCRIBEHOST_HOST: edgex-redis
    hostname: edgex-app-rules-engine
    image: nexus3.edgexfoundry.org:10004/app-service-configurable:latest
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:59701:59701/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: 2002:2001
  command:
    container_name: edgex-core-command
    depends_on:
      consul:
        condition: service_started
      database:
        condition: service_started
      metadata:
        condition: service_started
    environment:
      CLIENTS_CORE_COMMAND_HOST: edgex-core-command
      CLIENTS_CORE_DATA_HOST: edgex-core-data
      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
      CLIENTS_SUPPORT_NOTIFICATIONS_HOST: edgex-support-notifications
      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
      DATABASES_PRIMARY_HOST: edgex-redis
      EDGEX_SECURITY_SECRET_STORE: "false"
      MESSAGEQUEUE_HOST: edgex-redis
      REGISTRY_HOST: edgex-core-consul
      SERVICE_HOST: edgex-core-command
    hostname: edgex-core-command
    image: nexus3.edgexfoundry.org:10004/core-command:latest
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:59882:59882/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: 2002:2001
  consul:
    command: agent -ui -bootstrap -server -client 0.0.0.0
    container_name: edgex-core-consul
    hostname: edgex-core-consul
    image: consul:1.13
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:8500:8500/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: root:root
    volumes:
    - consul-config:/consul/config:z
    - consul-data:/consul/data:z
  data:
    container_name: edgex-core-data
    depends_on:
      consul:
        condition: service_started
      database:
        condition: service_started
      metadata:
        condition: service_started
    environment:
      CLIENTS_CORE_COMMAND_HOST: edgex-core-command
      CLIENTS_CORE_DATA_HOST: edgex-core-data
      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
      CLIENTS_SUPPORT_NOTIFICATIONS_HOST: edgex-support-notifications
      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
      DATABASES_PRIMARY_HOST: edgex-redis
      EDGEX_SECURITY_SECRET_STORE: "false"
      MESSAGEQUEUE_HOST: edgex-redis
      REGISTRY_HOST: edgex-core-consul
      SERVICE_HOST: edgex-core-data
    hostname: edgex-core-data
    image: nexus3.edgexfoundry.org:10004/core-data:latest
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:59880:59880/tcp
    - 0.0.0.0:5563:5563/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: 2002:2001
  database:
    container_name: edgex-redis
    environment:
      CLIENTS_CORE_COMMAND_HOST: edgex-core-command
      CLIENTS_CORE_DATA_HOST: edgex-core-data
      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
      CLIENTS_SUPPORT_NOTIFICATIONS_HOST: edgex-support-notifications
      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
      DATABASES_PRIMARY_HOST: edgex-redis
      EDGEX_SECURITY_SECRET_STORE: "false"
      MESSAGEQUEUE_HOST: edgex-redis
      REGISTRY_HOST: edgex-core-consul
    hostname: edgex-redis
    image: redis:7.0-alpine
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:6379:6379/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: root:root
    volumes:
    - db-data:/data:z
  metadata:
    container_name: edgex-core-metadata
    depends_on:
      consul:
        condition: service_started
      database:
        condition: service_started
      notifications:
        condition: service_started
    environment:
      CLIENTS_CORE_COMMAND_HOST: edgex-core-command
      CLIENTS_CORE_DATA_HOST: edgex-core-data
      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
      CLIENTS_SUPPORT_NOTIFICATIONS_HOST: edgex-support-notifications
      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
      DATABASES_PRIMARY_HOST: edgex-redis
      EDGEX_SECURITY_SECRET_STORE: "false"
      MESSAGEQUEUE_HOST: edgex-redis
      NOTIFICATIONS_SENDER: edgex-core-metadata
      REGISTRY_HOST: edgex-core-consul
      SERVICE_HOST: edgex-core-metadata
    hostname: edgex-core-metadata
    image: nexus3.edgexfoundry.org:10004/core-metadata:latest
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:59881:59881/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: 2002:2001
  notifications:
    container_name: edgex-support-notifications
    depends_on:
      consul:
        condition: service_started
      database:
        condition: service_started
    environment:
      CLIENTS_CORE_COMMAND_HOST: edgex-core-command
      CLIENTS_CORE_DATA_HOST: edgex-core-data
      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
      CLIENTS_SUPPORT_NOTIFICATIONS_HOST: edgex-support-notifications
      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
      DATABASES_PRIMARY_HOST: edgex-redis
      EDGEX_SECURITY_SECRET_STORE: "false"
      MESSAGEQUEUE_HOST: edgex-redis
      REGISTRY_HOST: edgex-core-consul
      SERVICE_HOST: edgex-support-notifications
    hostname: edgex-support-notifications
    image: nexus3.edgexfoundry.org:10004/support-notifications:latest
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:59860:59860/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: 2002:2001
  rulesengine:
    container_name: edgex-kuiper
    depends_on:
      database:
        condition: service_started
    environment:
      CONNECTION__EDGEX__REDISMSGBUS__PORT: 6379
      CONNECTION__EDGEX__REDISMSGBUS__PROTOCOL: redis
      CONNECTION__EDGEX__REDISMSGBUS__SERVER: edgex-redis
      CONNECTION__EDGEX__REDISMSGBUS__TYPE: redis
      EDGEX__DEFAULT__PORT: 6379
      EDGEX__DEFAULT__PROTOCOL: redis
      EDGEX__DEFAULT__SERVER: edgex-redis
      EDGEX__DEFAULT__TOPIC: rules-events
      EDGEX__DEFAULT__TYPE: redis
      KUIPER__BASIC__CONSOLELOG: "true"
      KUIPER__BASIC__RESTPORT: 59720
    hostname: edgex-kuiper
    image: lfedge/ekuiper:1.6-alpine
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:59720:59720/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: kuiper:kuiper
    volumes:
    - kuiper-data:/kuiper/data:z
  scheduler:
    container_name: edgex-support-scheduler
    depends_on:
      consul:
        condition: service_started
      database:
        condition: service_started
    environment:
      CLIENTS_CORE_COMMAND_HOST: edgex-core-command
      CLIENTS_CORE_DATA_HOST: edgex-core-data
      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
      CLIENTS_SUPPORT_NOTIFICATIONS_HOST: edgex-support-notifications
      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
      DATABASES_PRIMARY_HOST: edgex-redis
      EDGEX_SECURITY_SECRET_STORE: "false"
      INTERVALACTIONS_SCRUBAGED_HOST: edgex-core-data
      INTERVALACTIONS_SCRUBPUSHED_HOST: edgex-core-data
      MESSAGEQUEUE_HOST: edgex-redis
      REGISTRY_HOST: edgex-core-consul
      SERVICE_HOST: edgex-support-scheduler
    hostname: edgex-support-scheduler
    image: nexus3.edgexfoundry.org:10004/support-scheduler:latest
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:59861:59861/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: 2002:2001
  system:
    container_name: edgex-sys-mgmt-agent
    depends_on:
      command:
        condition: service_started
      consul:
        condition: service_started
      data:
        condition: service_started
      metadata:
        condition: service_started
      notifications:
        condition: service_started
      scheduler:
        condition: service_started
    environment:
      CLIENTS_CORE_COMMAND_HOST: edgex-core-command
      CLIENTS_CORE_DATA_HOST: edgex-core-data
      CLIENTS_CORE_METADATA_HOST: edgex-core-metadata
      CLIENTS_SUPPORT_NOTIFICATIONS_HOST: edgex-support-notifications
      CLIENTS_SUPPORT_SCHEDULER_HOST: edgex-support-scheduler
      DATABASES_PRIMARY_HOST: edgex-redis
      EDGEX_SECURITY_SECRET_STORE: "false"
      EXECUTORPATH: /sys-mgmt-executor
      MESSAGEQUEUE_HOST: edgex-redis
      METRICSMECHANISM: executor
      REGISTRY_HOST: edgex-core-consul
      SERVICE_HOST: edgex-sys-mgmt-agent
    hostname: edgex-sys-mgmt-agent
    image: nexus3.edgexfoundry.org:10004/sys-mgmt-agent:latest
    networks:
      edgex-network: null
    ports:
    - 0.0.0.0:58890:58890/tcp
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    - label:disable
    user: root:root
    volumes:
    - /var/run/docker.sock:/var/run/docker.sock:z
  ui:
    container_name: edgex-ui-go
    environment:
      EDGEX_SECURITY_SECRET_STORE: "false"
    hostname: edgex-ui-go
    image: nexus3.edgexfoundry.org:10004/edgex-ui:latest
    networks:
      edgex-network: null
    ports:
    - published: 4000
      target: 4000
    read_only: true
    restart: always
    security_opt:
    - no-new-privileges:true
    user: 2002:2001
version: '3.7'
volumes:
  consul-config: {}
  consul-data: {}
  db-data: {}
  kuiper-data: {}

**Reply by lenny-goodell** [reply]
Please upload as an attachment so YAML indentions are preserved.

**Reply by sounak-1989** [reply]
I am unable to upload *.yml file in discussions. Github discussions is not letting me.

**Reply by lenny-goodell** [reply]
Change the extension to .txt or something it likes.

**Reply by sounak-1989** [reply]
[docker-compose.txt](https://github.com/edgexfoundry/community/files/9760156/docker-compose.txt)

**Reply by lenny-goodell** [reply]
All services run fine for me using your compose file. I am running in WSL2, Ubuntu 18.04, Docker 20.10.8 and docker-compose 1.27.2

**Reply by sounak-1989** [reply]
Hmm, let me check the port forwarding for my VirtualBox. Do you think that might be a problem?

**Reply by lenny-goodell** [reply]
No, since this communication is all in the docker network running in your Ubuntu on  your VirtualBox

**Reply by sounak-1989** [reply]
I finally gave up on Ubuntu and did a fresh install of Debian. Works great.

**Reply by lenny-goodell** [reply]
@sounak-1989 , thanks for the update. 
@farshidtz, any thoughts on why there are issue using `Ubuntu Server 22.04 LTS in a VirtualBox inside Windows 10` ? See comments above about docker networking issue. THX?
